src/contracts/LiquidationEngine.sol

```diff
@@ -175,10 +175,16 @@ contract LiquidationEngine is
 
         if (_collateralToSell == 0) revert LiqEng_NullCollateralToSell();
 
+        if (currentOnAuctionSystemCoins + _limitAdjustedDebt > _params.onAuctionSystemCoinLimit) {
+          revert LiqEng_ExceededCapacity();
+        }
+
         if (
           _limitAdjustedDebt != _safeData.generatedDebt
             && (_safeData.generatedDebt - _limitAdjustedDebt) * _safeEngCData.accumulatedRate < _debtFloor
-        ) revert LiqEng_DustySAFE();
+        ) {
+          _limitAdjustedDebt = _safeData.generatedDebt;
```

Comment:
Previous variables such as `_collateralToSell` and `_amountToRaise` are computed based on the first `_limitAdjustedDebt` value.

Reply:
hmm we should need to recompute them? letme see.

Reply:
ok, so, this is not a 'health check' anymore, let's bring this logic a few lines up, so the other variables are updated accordingly 🙏